cmd/driver-registrar/main.go

```diff
@@ -50,14 +55,60 @@ const (
 
 // Command line flags
 var (
-	kubeconfig        = flag.String("kubeconfig", "", "Absolute path to the kubeconfig file. Required only when running out of cluster.")
-	connectionTimeout = flag.Duration("connection-timeout", 1*time.Minute, "Timeout for waiting for CSI driver socket.")
-	csiAddress        = flag.String("csi-address", "/run/csi/socket", "Address of the CSI driver socket.")
-	showVersion       = flag.Bool("version", false, "Show version.")
-
-	version = "unknown"
+	kubeconfig              = flag.String("kubeconfig", "", "Absolute path to the kubeconfig file. Required only when running out of cluster.")
+	connectionTimeout       = flag.Duration("connection-timeout", 1*time.Minute, "Timeout for waiting for CSI driver socket.")
+	csiAddress              = flag.String("csi-address", "/run/csi/socket", "Address of the CSI driver socket.")
+	kubeletRegistrationPath = flag.String("kubelet-registration-path", "/var/lib/kubelet/plugins/csi-hostpath/csi.sock",
+		`Enables Kubelet Plugin Registration service, and returns the specified path as "endpoint" in "PluginInfo" response.
+		 If this option is set, the driver-registrar expose a unix domain socket to handle Kubelet Plugin Registration, 
+		 this socket MUST be surfaced on the host in the kubelet plugin registration directory (in addition to the CSI driver socket). 
+		 If plugin registration is enabled on kubelet (kubelet flag KubeletPluginsWatcher is set), then this option should be set
+		 and the value should be the path of the CSI driver socket on the host machine.`)
+	showVersion = flag.Bool("version", false, "Show version.")
+	version     = "unknown"
+	// List of supported versions
+	supportedVersions = []string{"0.2.0", "0.3.0"}
 )
 
+// registrationServer is a sample plugin to work with plugin watcher
+type registrationServer struct {
+	driverName string
+	endpoint   string
+	version    []string
+}
+
+var _ registerapi.RegistrationServer = registrationServer{}
+
+// NewregistrationServer returns an initialized registrationServer instance
+func newRegistrationServer(driverName string, endpoint string, versions []string) *registrationServer {
```

Comment:
nit: `new` constructor methods should return interfaces instead of a concrete types (makes testing easier). So change this to:

```GO
func newRegistrationServer(driverName string, endpoint string, versions []string) registerapi.RegistrationServer {
```